	* configure.in: Add $(EXEEXT) as necessary for Win32 platforms.

--- origsrc/genius-1.0.6/configure.in	2009-02-27 14:39:50.000000000 -0600
+++ src/genius-1.0.6/configure.in	2009-03-29 13:07:02.328125000 -0500
@@ -150,8 +150,8 @@ Note:
   VICIOUS_UI_LIBS="libviciousui.a"
   GTKEXTRA_LIBS="libgtkextra-genius.a"
   
-  GNOME_GENIUS=gnome-genius
-  GENIUS_READLINE_HELPER_FIFO=genius-readline-helper-fifo
+  GNOME_GENIUS="gnome-genius\$(EXEEXT)"
+  GENIUS_READLINE_HELPER_FIFO="genius-readline-helper-fifo\$(EXEEXT)"
   GNOME_GENIUS_DESKTOP_IN="gnome-genius.desktop.in"
 
 else
