--- origsrc/genius-1.0.8/src/gnome-genius.c	2009-11-10 00:10:57.000000000 -0600
+++ src/genius-1.0.8/src/gnome-genius.c	2009-11-12 01:24:49.326201200 -0600
@@ -1637,9 +1637,7 @@ gel_printout_infos (void)
 static void
 actually_open_help (const char *id)
 {
-/* breaks binary back compatibility */
-#if 0
-/*#if GTK_CHECK_VERSION(2,14,0) */
+#if GTK_CHECK_VERSION(2,14,0)
 	GError *error = NULL;
 	char *str;
 
@@ -1660,8 +1658,7 @@ actually_open_help (const char *id)
 		g_free (str);
 		g_error_free (error);
 	}
-/*#else*/
-#endif
+#else
 	char *xdgopen;
 	char *uri;
 	char *file = NULL;
@@ -1720,7 +1717,7 @@ actually_open_help (const char *id)
 
 	g_free (xdgopen);
 	g_free (uri);
-/*#endif*/
+#endif
 }
 
 void
