--- origsrc/genius-1.0.3/configure.in	2008-02-22 02:11:30.000000000 -0600
+++ src/genius-1.0.3/configure.in	2008-06-17 00:30:30.671875000 -0500
@@ -152,8 +152,8 @@
   VICIOUS_UI_LIBS="libviciousui.a"
   GTKEXTRA_LIBS="libgtkextra-genius.a"
   
-  GNOME_GENIUS=gnome-genius
-  GENIUS_READLINE_HELPER_FIFO=genius-readline-helper-fifo
+  GNOME_GENIUS='gnome-genius$(EXEEXT)'
+  GENIUS_READLINE_HELPER_FIFO='genius-readline-helper-fifo$(EXEEXT)'
   GNOME_GENIUS_DESKTOP_IN="gnome-genius.desktop.in"
 
 else
--- origsrc/genius-1.0.3/src/Makefile.am	2008-05-19 23:53:33.000000000 -0500
+++ src/genius-1.0.3/src/Makefile.am	2008-06-17 01:45:29.578125000 -0500
@@ -165,6 +165,8 @@
 	geloutput.h	\
 	funclibhelper.cP
 
+genius_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libgenius.a
+
 genius_LDADD = \
 	$(top_builddir)/ve/libvicious.a		\
 	-lm					\
@@ -193,7 +195,9 @@
 
 plugin_LTLIBRARIES = libtestplugin.la
 
-libtestplugin_la_LDFLAGS = -module -avoid-version
+libtestplugin_la_DEPENDENCIES = genius$(EXEEXT)
+libtestplugin_la_LDFLAGS = -module -avoid-version -no-undefined -rpath $(pkglibdir) -Wl,libgenius.a
+libtestplugin_la_LIBADD = $(INTLLIBS)
 
 libtestplugin_la_SOURCES =	\
 	plug_api.h		\
@@ -238,3 +242,6 @@
 	sed -e 's,\@libdir\@,$(libdir),g' < $(srcdir)/test.plugin.in \
 	  > test.plugin.tmp && mv -f test.plugin.tmp test.plugin
 
+install-exec-hook:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libgenius.a $(DESTDIR)$(libdir)
--- origsrc/genius-1.0.3/ve/Makefile.am	2007-11-17 21:25:58.000000000 -0600
+++ src/genius-1.0.3/ve/Makefile.am	2008-06-17 00:29:01.890625000 -0500
@@ -45,7 +45,7 @@
 	test-ve-config.c
 
 test_ve_config_LDADD = \
-	$(top_builddir)/ve/libvicious.a		\
+	libvicious.a		\
 	$(VICIOUS_LIBS)				\
 	$(INTLLIBS)				\
 	$(GLIB_LIBS)
